src/commands/publish/preview/mod.rs

```diff
@@ -69,6 +60,36 @@ pub fn preview(
     Ok(())
 }
 
+fn upload_to_preview(
+    project: &Project,
+    user: Option<GlobalUser>,
+    client: &reqwest::Client,
+) -> Result<Preview, failure::Error> {
+    let create_address = match &user {
+        Some(_user) => format!(
+            "https://api.cloudflare.com/client/v4/accounts/{}/workers/scripts/{}/preview",
+            project.account_id, project.name
+        ),
+        None => "https://cloudflareworkers.com/script".to_string(),
+    };
+
+    let script_upload_form = publish::build_script_upload_form(project)?;
+
+    let mut res = client
```

Comment:
I'm guessing this is where headers logic (for the passing of email and the api token) will live :)

Reply:
We'll probably need two cases of upload call--one for auth, one for not?

Reply:
Oooh, does the cf apiclient create allow us to call the v4 api here (endpoint https://api.cloudflare.com/client/v4/accounts/{}/workers/scripts/{}/preview)? I think that would be super useful to use.

Reply:
the only differences are in the url we use, and in the auth headers. auth headers are set in the client (preview constructs either an auth client or a regular client), and the url is set above (create_address). If the naming is confusing tho we should fix that.

Reply:
> Oooh, does the cf apiclient crate allow us to call the v4 api here (endpoint [https://api.cloudflare.com/client/v4/accounts/{}/workers/scripts/{}/preview](https://api.cloudflare.com/client/v4/accounts/%7B%7D/workers/scripts/%7B%7D/preview))? I think that would be super useful to use.

i think i know what you're asking; if by that you mean https://github.com/cloudflare/cloudflare-rs, eventually i want to add workers api support there as well and make it our de-facto v4 api client, but i'm not there yet.

Reply:
^ You're correct about what I'm asking! We can save a migration to the cloudflare-rs client for later; this could be worth mentioning in a `//todo` comment. I also now see how the headers are set in the client that's passed into upload_to_preview--I think we can leave that untouched for now. :)

Reply:
the cloudflare-rs client migration was one of my Q3 goals; i'm inclined to write an issue for it rather than a comment.